content/tutorials/git_authentication/index.Rmd

```diff
@@ -0,0 +1,266 @@
+
+## <a name="ssh"></a>Creating public-private SSH-key pairs
+
+Below we describe the steps you need to run in order to make SSH-keys (Secure SHell).
+The SSH protocol is, next to the HTTPS-protocol, a way to safely interact with 
+GitHub.
+A general and easy to follow explanation about SSH can be read [here](https://dev.to/risafj/ssh-key-authentication-for-absolute-beginners-in-plain-english-2m3f).
+Both HTTPS and SSH are secure ways to communicate with a server (pass information between your computer and a server).
+For a discussion of technical differences and similarities on how both protocols handle security, [this resouce about SSH and TLS](https://www.ssl2buy.com/wiki/ssh-vs-ssl-tls) is useful.
+
+
+A private key, that is specific to your computer, and a matching public key that
+is stored on your GitHub account are needed.
+The provided instructions are for Windows and are taken from
+https://happygitwithr.com/ssh-keys.html.
+The same source may be consulted for other operating systems.
+
+1. Open RStudio: _Tools > Global Options…> Git/SVN > Create RSA Key..._ You can
+optionally use a passphrase for extra protection of the key.
+Without password, anyone who has a copy of your private key, can impersonate you when authenticating on GitHub.
+When the private key is password protected they need your password too.
+Click create and apply.
+
+2. In RStudio, open a Rproject with Git version control and navigate to the Git
+pane.
+Open the Git Shell (_More > Shell_).
+First we check if the SSH agent works with the following command
+(The first $ is the prompt and you should not copy it. To paste in the shell
+click the right mouse button.):
+
+    ```
+    $ eval $(ssh-agent -s)
+    # which should give something like this:
+    Agent pid 59566
+    ```
+
+    If that works, we need to add the SSH key (if you provided a passphrase, you
+    will need it here.):
+
+    ```
+    $ ssh-add ~/.ssh/id_rsa
+    ```
+
+3. Restart RStudio. Next _Tools > Global Options…> Git/SVN_ and click
+_View public key_ in the SSH-RSA section.
+Copy to the clipboard.
+
+4. Open your personal GitHub account _https://github.com/<githubusername>_
+(you may need to login). 
+Click your profile-icon in the upperright corner and go to
+_Settings > SSH and GPG keys_.
+Click _New SSH key_.
+Paste the public key in the field and provide an informative title
+(e.g. <year>-<computername>).
+Click _Add SSH Key_.
+
+5. By default communication with GitHub is via the HTTPS-protocol.
+To use the SSH protocol, you need to set the remote of your repository correctly.
```

Comment:
That equally holds for HTTPS :slightly_smiling_face:.

Reply:
See [29870b0](https://github.com/inbo/tutorials/pull/309/commits/29870b07f5cb14868411f5b036dc9575d7b92ba9)